EKF/mag_control.cpp

```diff
@@ -327,3 +329,12 @@ void Ekf::run3DMagAndDeclFusions()
 	}
 }
 
+bool Ekf::otherHeadingSourcesHaveFinished()
+{
+    // detect rising edge
+    bool result = noOtherYawAidingThanMag() && !_no_non_mag_heading_aiding_running;
+
+    _no_non_mag_heading_aiding_running = noOtherYawAidingThanMag();
```

Comment:
```suggestion
    bool result = noOtherYawAidingThanMag() && _non_mag_yaw_aiding_running_prev;

    _non_mag_yaw_aiding_running_prev = !noOtherYawAidingThanMag();
```

Reply:
using the same name as the function and not using double negation makes it simpler to read IMO. (also, "heading_aiding" is a bit strange to pronounce :D)